### PR TITLE
[android] #5211 - run UI tests in our CI workflow

### LINKED ISSUE #5211: Android integration tests
In order to prevent regressions like #5108, we need Android integration tests that test actually running a test app.

### PR DESCRIPTION
**Pull request by tobrun**
**tl;dr**: Integrates Google Firebase device lab into Bitrise

Closes #5211, For each commit pushed to this repo, our CI will run tests on Firebase device lab. 

Benefit of his approach is that these tests are deployed on a real device but as downside it also increases the android build time from 10 -> 17 minutes with 4 minutes of actual device testing. 

This PR includes:
 - added make target for generating debug and test apks
 - auth with Google service account credentials from bitrise configuration
 - auth with dev map accesstoken from bitrise configuration
 - runs tests on firebase device lab
 - fails the build a test test fails
 - ignored some tests that were failing
 - excluded locational test activities from test generation

Review @ivovandongen / @zugaldia ?

### CODE REVIEW

**Comment by tobrun** [comment]
The build is still faster as the Travis build 👍 

screen shot 2016-08-18 at 15 06 22

File: platform/android/scripts/generate-test-code.js [hunk 1]
```diff
@@ -53,6 +54,8 @@ for(const subPackage of subPackages) {
       if (!(excludeActivities.indexOf(activityName) > -1)) {
         console.log("Created file:  "+filePath);
         fs.writeFileSync(filePath, ejsConversionTask([activityName, subPackage]));
+      }else{
```
Comment (ivovandongen) [inline platform/android/scripts/generate-test-code.js]: Nit: small spacing issue

**Comment by ivovandongen** [comment]
@tobrun Looks great, still runs locally. Go for it!

**Comment by tobrun** [comment]
> still runs locally

It seems you were surprised about that :trollface:

**Comment by tobrun** [comment]
CI has approved, @ivovandongen confirmed, time to press the green button!

**Comment by tobrun** [comment]
cc @mapbox/android

**Comment by zugaldia** [comment]
> CI has approved

Of course CI has approved, CI should be very happy about this PR! :)
